scripts/retrieve_tyndp_bundle.py

```diff
@@ -0,0 +1,86 @@
+# SPDX-FileCopyrightText: 2024 The PyPSA-Eur Authors
+#
+# SPDX-License-Identifier: MIT
+"""
+.. image:: https://zenodo.org/badge/DOI/10.5281/zenodo.14230568.svg
+  :target: https://doi.org/10.5281/zenodo.14230568
+
+The data bundle contains input data for the 2024 TYNDP scenario building process.
+
+This rule downloads the TYNDP data bundle from `zenodo
+<https://doi.org/10.5281/zenodo.14230568>` and extracts it in the ``data``
+subdirectory, such that all files of the bundle are stored in the
+``data/tyndp_2024_bundle`` subdirectory.
+
+**Outputs**
+
+- ``data/tyndp_2024_bundle``: input data for TYNDP 2024 scenario building
```

Comment:
Needs to be updated to match revised URLs.

Please also add this new data source to `date-sources.rst`.